packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -1634,3 +1634,159 @@ test.describe('binding_property_non_reactive warn', () => {
 		expect(is_warning_thrown).toBeFalsy();
 	});
 });
+
+test.describe('navigation and redirects should be consistent between web native and sveltekit based', () => {
+	function expectPath(page) {
+		return expect(new URL(page.url()).pathname);
+	}
+
+	test.beforeEach(async ({ page }) => {
+		await page.goto('/navigation/teststart');
+		expectPath(page).toBe('/navigation/teststart');
+	});
+
+	test.describe('navigating outside the app on sameorigin', () => {
+		[
+			[
+				'app.goto { replaceState: false }',
+				async ({ app }) => {
+					await app.goto('/navigation/nonexistent', { replaceState: false }).catch(() => {});
+				}
+			],
+			[
+				'location.assign',
+				async ({ page }) => {
+					await page.evaluate(() => {
+						location.assign('/navigation/nonexistent');
+					});
+				}
+			]
+		].forEach(([title, navigate]) => {
+			test(title, async ({ app, page }) => {
+				await navigate({ app, page });
+				await page.waitForLoadState('networkidle');
+				expectPath(page).toBe('/navigation/nonexistent');
+
+				await page.goBack();
+				expectPath(page).toBe('/navigation/teststart');
+			});
+		});
+
+		[
+			[
+				'app.goto { replaceState: true }',
+				async ({ app }) => {
+					await app.goto('/navigation/nonexistent', { replaceState: true }).catch(() => {});
+				}
+			],
+			[
+				'location.replace',
+				async ({ page }) => {
+					await page.evaluate(() => {
+						location.replace('/navigation/nonexistent');
+					});
+				}
+			]
```

Comment:
I don't think we need this test either since it's only testing the browser functionality and nothing on SvelteKit's side.